plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerPlugin.java

```diff
@@ -140,119 +137,138 @@ public void setLocationLayerEnabled(boolean isEnabled) {
     }
   }
 
-  private void enableLocationLayerPlugin() {
-    isEnabled = true;
-
-    if (locationEngine != null) {
-      locationEngine.addLocationEngineListener(this);
-    }
-    setLastLocation();
-    setLastCompassHeading();
-    locationLayer.show();
-  }
-
-  private void disableLocationLayerPlugin() {
-    isEnabled = false;
-
-    if (locationEngine != null) {
-      locationEngine.removeLocationEngineListener(this);
-    }
-    locationLayer.hide();
-  }
-
   /**
-   * After creating an instance of this plugin, you can use this API to enable the location mode of
-   * your choice. These modes can be found in the {@link LocationLayerMode} class and the parameter
-   * only accepts one of those modes. Note that before enabling the My Location layer, you will need
-   * to ensure that you have the requested the required user location permissions.
+   * Sets the camera mode, which determines how the map camera will track the rendered location.
    * <p>
    * <ul>
-   * <li>{@link LocationLayerMode#NORMAL}: Display the user location on the map as a small dot</li>
-   * <li>{@link LocationLayerMode#COMPASS}: Display the user location and current heading/bearing</li>
-   * <li>{@link LocationLayerMode#NAVIGATION}: Display the user location on the map using a navigation icon</li>
+   * <li>{@link CameraMode#NONE}: No camera tracking</li>
+   * <li>{@link CameraMode#NONE_COMPASS}: Camera does not track location, but does track compass bearing</li>
+   * <li>{@link CameraMode#NONE_GPS}: Camera does not track location, but does track GPS bearing</li>
+   * <li>{@link CameraMode#TRACKING}: Camera tracks the user location</li>
+   * <li>{@link CameraMode#TRACKING_COMPASS}: Camera tracks the user location, with bearing provided by a compass</li>
+   * <li>{@link CameraMode#TRACKING_GPS}: Camera tracks the user location, with normalized bearing</li>
+   * <li>{@link CameraMode#TRACKING_GPS_NORTH}: Camera tracks the user location, with bearing always set to north</li>
    * </ul>
    *
-   * @param locationLayerMode one of the modes found in {@link LocationLayerMode}
-   * @since 0.1.0
+   * @param cameraMode one of the modes found in {@link CameraMode}
+   * @since 0.5.0
    */
   public void setCameraMode(@CameraMode.Mode int cameraMode) {
     locationLayerCamera.setCameraMode(cameraMode);
   }
 
+  /**
+   * Provides the current camera mode being used to track
+   * the location or compass updates.
+   *
+   * @return the current camera mode
+   * @since 0.5.0
+   */
   @CameraMode.Mode
   public int getCameraMode() {
     return locationLayerCamera.getCameraMode();
   }
 
+  /**
+   * Sets the render mode, which determines how the location updates will be rendered on the map.
+   * <p>
+   * <ul>
+   * <li>{@link RenderMode#NORMAL}: Basic tracking is enabled, bearing ignored</li>
+   * <li>{@link RenderMode#COMPASS}: Tracking the user location with bearing considered from compass</li>
```

Comment:
Can we change `tracking` keyword from `setRenderMode()` doc to just say something along the lines of "show current user location", not to confuse it with the camera tracking? It should be implicit that it'll always be the most recent location sample available.